pkg/controller/namespace/namespace_controller_utils.go

```diff
@@ -385,6 +390,11 @@ func syncNamespace(
 		return err
 	}
 
+	// the latest view of the namespace asserts that namespace is no longer deleting..
+	if namespace.DeletionTimestamp.IsZero() {
```

Comment:
given the current code structure, we do the following:

sync loop:

1. get latest view of a namespace
2. update the status if deletion timestamp is non-zero, else do nothing
3. if the do-nothing flow happened, then we need to check for a deletion timestamp here...

so i think this should stay in the current spot for now.